### PR TITLE
[DEVOPS-318] Add approval workflow

### PR DESCRIPTION
**Pull request by FleetAdmiralButter**
1) Add parameter to reference the environment passed in by the caller repository so that an approval workflow can be applied.

### CODE REVIEW

File: .github/workflows/tf_apply.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@ on:
         type: string
         required: false
         default: ap-southeast-2
+      environment:
+        description: GitHub environment for reviewer approval
+        type: string
+        required: false
```
Comment (GROwen) [inline .github/workflows/tf_apply.yml:13]: @FleetAdmiralButter Can you point me to the AC for this please, I'd like to be able to determine if `production` is a sane default.

Comment (FleetAdmiralButter) [reply .github/workflows/tf_apply.yml:13]: Hey @GROwen, it's from this ticket: https://digital-vic.atlassian.net/browse/DDS-1577

**Review by GROwen — APPROVED** [review]

**Comment by nicksantamaria** [comment]
Hey @FleetAdmiralButter I don't really understand how the new `environment` variable adds an approval step. Could you add some more details?

**Comment by FleetAdmiralButter** [comment]
Hi @nicksantamaria, approval steps in GitHub Actions are tied to environments - https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments#about-required-reviews-in-workflows.

On the internal Terraform repository, I've created an environment with a list of approvers. The `environment` variable here ties the workflow to a specific environment configured on the repository, which allows it to enforce approval of the workflow (via the environment's deploy rules) before it starts.

**Comment by steveworley** [comment]
Gonna merge this one to test the approval in AKS extras, I'll link the relevant workflow to this PR.

**Comment by steveworley** [comment]
@nicksantamaria it integrates with the standard Github UI and workflows ends up looking like this:

image

It tracks the approvals and does it all via the Github UI.

**Pull request merged**